lua/lint/util.lua

```diff
@@ -21,5 +21,25 @@ function M.offset_to_position(lines, offset)
   }
 end
 
+-- Returns the path leading up to (and including) the given directory, based on
+-- the current buffer's file path.
+--
+-- Example:
+--
+-- The buffer path is `foo/bar/baz.txt`. When calling this function with the
+-- second argument set to `bar`, this function returns `foo/bar`.
+function M.find_nearest_directory(directory)
```

Comment:
This is ported from ALE. The Inko linter uses this to find the root "tests" directory for a given buffer. I saw the other utli function has some tests, but I'm not sure how to add tests for this function :confused: